torch/distributed/_shard/sharding_spec/chunk_sharding_spec.py

```diff
@@ -62,7 +62,7 @@ def __post_init__(self):
                 self.placements[i] = torch.distributed._remote_device(remote_device)
 
     @staticmethod
-    def _verify_dim(dim):
+    def _verify_dim(dim) -> None:
```

Comment:
Use the same overload trick to tell the type checker this narrows the arg type.

Reply:
This fix is too complicated that I would rather not typing it.